Backend/src/middlewares/logger.ts

```diff
@@ -0,0 +1,59 @@
+import winston, { format } from "winston";
+import * as expressWinston from "express-winston";
+import path from "path"
+
+let requestLoggerTransports: Array<any> = [
+    //always save to file
+    new winston.transports.File({ filename: path.join(process.cwd(), "logs", "requests.log") })
+]
+let errorLoggerTransports: Array<any> = [
+    //always save to file
+    new winston.transports.File({ filename: path.join(process.cwd(), "logs", "error.log") })
+]
+if (process.env.NODE_ENV !== 'production') {
+    //if dev, also log to console
+    requestLoggerTransports.push(new winston.transports.Console({
+        format: format.combine(
+            //format.colorize({ all: true }),
+            format.json(),
+            format.simple(),
+            format.prettyPrint()
+            //format.splat()
+        ),
+    }));
+    errorLoggerTransports.push(new winston.transports.Console({
+        format: format.combine(
+            //format.colorize({ all: true }),
+            format.json(),
+            //format.errors({ stack: true }), //print stacktrace to console
+            format.simple(),
+            format.prettyPrint()
+        )
+    }));
+}
```

Comment:
This means that it is _only_ logging in Production environments?

Reply:
Only in development do we also log to console. When the application is deployed we only log to file, as is the default.